binding/java/src/ai/picovoice/porcupine/Utils.java

```diff
@@ -121,24 +121,78 @@ private static Path extractResources(Path jarPath) throws IOException {
+        if (os.contains("mac") || os.contains("darwin")) {
+            return "mac";
+        } else if (os.contains("win")) {
+            return "windows";
+        } else if (os.contains("linux")) {
+            String arch = System.getProperty("os.arch");
+            if (arch.equals("arm") || arch.equals("aarch64")) {
+                String cpuPart = getCpuPart();
+                switch (cpuPart) {
+                    case "0xc07":
+                    case "0xd03":
+                    case "0xd08":
+                        return "raspberry-pi";
+                    case "0xd07":
+                        return "jetson";
+                    case "0xc08":
+                        return "beaglebone";
+                    default:
+                        throw new RuntimeException(String.format("Execution environment not supported. " +
+                                "Porcupine Java does not support CPU Part (%s).", cpuPart));
+                }
+            }
+            return "linux";
+        } else {
+            throw new RuntimeException("Execution environment not supported. " +
+                    "Porcupine Java is supported on MacOS, Linux and Windows");
+        }
+    }
+
+    private static String getArchitecture() throws RuntimeException {
         String arch = System.getProperty("os.arch");
         if (arch.equals("amd64") || arch.equals("x86_64")) {
-            String os = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
-            if (os.contains("mac") || os.contains("darwin")) {
-                return "mac";
-            } else if (os.contains("win")) {
-                return "windows";
-            } else if (os.contains("linux")) {
-                return "linux";
+            if (ENVIRONMENT_NAME.equals("windows")) {
+                return "amd64";
             } else {
-                logger.severe("Execution environment not supported. " +
-                        "Porcupine Java is supported on MacOS, Linux and Windows");
-                return null;
+                return "x86_64";
+            }
+        } else if (arch.equals("arm") || arch.equals("aarch64")) {
+            String cpuPart = getCpuPart();
+            String archInfo = (arch.equals("aarch64")) ? "-aarch64" : "";
+
+            switch (cpuPart) {
+                case "0xc07":
+                    return "cortex-a7" + archInfo;
+                case "0xd03":
+                    return "cortex-a53" + archInfo;
+                case "0xd07":
+                    return "cortex-a57" + archInfo;
+                case "0xd08":
+                    return "cortex-a72" + archInfo;
+                case "0xc08":
+                    return "";
+                default:
+                    throw new RuntimeException(
+                            String.format("Platform architecture with CPU Part (%s) is not supported by Porcupine.", cpuPart)
```

Comment:
don't we usually revert to some generic arm build at this point instead of throwing a fatal?